app/views/shared/_header.html.haml

```diff
@@ -1,22 +1,23 @@
 %section#top-header.row
   %h1.col-md-2
     = link_to "MyFLiX", root_path
-  %ul.col-md-4.clearfix
-    %li= link_to "Videos"
-    %li= link_to "My Queue"
-    %li= link_to "People"
-  %form.col-md-5.navbar-form(action="/videos/search" method="get")
-    .form-group
-      %input.form-control(type="text" name="q" placeholder="Search for videos here")
-    %button.btn.btn-default(type="submit") Search
-  #user_links.pull-right
-    %ul
-      %li.dropdown
-        %a(href="#" id="dlabel" role="button" data-toggle="dropdown" class="dropdown-toggle" data-target="#")
-          Welcome, Philip J. Fry
-          %b.caret
-        %ul.dropdown-menu(role="menu" aria-labelledby="dlabel")
-          %li
-            %a(href="#") Account
-            %a(href="#") Plan and Billing
-            %a(href="#") Sign Out
+  - if logged_in?
+    %ul.col-md-4.clearfix.list-unstyled.list-inline#my_options
+      %li= link_to "Videos"
+      %li= link_to "My Queue"
+      %li= link_to "People"
+    %form.col-md-5.navbar-form(action="/videos/search" method="get")
```

Comment:
You should aim for Rails helpers as they provide to methods additional stuff, like CSRF token for extra security. Also its more readable.

My second complaint is that you are using hardcoded action path. You should always use named routes.